conn2res/reservoir.py

```diff
@@ -445,7 +445,10 @@ def simulate(self, Vext, ic=None, mode='forward'):
                 # get voltage at internal nodes
                 Vi.append(self.iterate(Ve))
         
-        return np.asarray(Vi)
+        V = np.zeros((len(Vi), self._n_nodes))
+        V[:, self._I] = np.asarray(Vi)
+        V[:, self._E] = Vext
+        return V
```

Comment:
`EchoStateNetwork` returns a matrix that is size (n_times, n_nodes), while `MemristiveReservoir` was returning (n_times, n_internal_nodes). This messes up the indexing for extracting readout nodes, for example, because those indexes were created relative to all nodes that exist, not just internal nodes.